crates/tabby-scheduler/src/code/index.rs

```diff
@@ -130,3 +130,23 @@ fn is_valid_file(file: &SourceCode) -> bool {
     file.max_line_length <= MAX_LINE_LENGTH_THRESHOLD
         && file.avg_line_length <= AVG_LINE_LENGTH_THRESHOLD
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_code_splitter() {
+        let intelligence = CodeIntelligence::default();
+        let file_contents = include_str!("../../../http-api-bindings/src/chat/openai_chat.rs");
```

Comment:
add insta test (like what we do for chat / completion golden test) for selected files.